test/e2e/performanceprofile/functests/6_mustgather_testing/mustgather.go

```diff
@@ -166,8 +172,17 @@ var _ = Describe("[rfe_id: 50649] Performance Addon Operator Must Gather", Label
 func checkfilesExist(listOfFiles []string, path string) error {
 	for _, f := range listOfFiles {
 		file := filepath.Join(path, f)
-		if _, err := os.Stat(file); errors.Is(err, os.ErrNotExist) {
-			return err
+		pattern, err := filepath.Glob(file)
+		if err != nil {
+			// if the pattern itself is malformed.
+			return fmt.Errorf("failed to match pattern for file: %s, error: %w", file, err)
+		}
+		if pattern == nil {
+			// When no files matched the pattern.
+			return fmt.Errorf("no files matched the pattern for file: %s", file)
+		}
+		if _, err = os.Stat(pattern[0]); errors.Is(err, os.ErrNotExist) {
```

Comment:
you are not covering the case when error is not nil but also not  os.ErrNotExist

Reply:
Resolved.